integration-tests/features/steps/common.py

```diff
@@ -943,3 +943,54 @@ def stack_analysis_check_companion_packages(json_data):
         assert companion_package not in analyzed_packages, \
             "The analyzed package '%s' is found in companion packages as well" \
             % companion_package
+
+
+def replaces_component(replacement, component, version):
+    assert "replaces" in replacement
+    replaces = replacement["replaces"]
+    for replace in replaces:
+        assert "name" in replace
+        assert "version" in replace
+        if replace["name"] == component and replace["version"] == version:
+            return True
+    return False
+
+
+def find_replacements(alternates, component, version):
+    return [replacement
+            for replacement in alternates
+            if replaces_component(replacement, component, version)]
+
+
+@then('I should find that the component {component} version {version} can be replaced by component {replaced_by} version {replacement_version}')
+def stack_analysis_check_replaces(json_data, component, version, replaced_by, replacement_version):
+    """Check that the component is replaced by the given package
+       and version."""
+    json_data = context.response.json()
+    path = "result/0/recommendations/alternate"
+    alternates = get_value_using_path(json_data, path)
+    replacements = find_replacements(alternates, component, version)
+
+    for replacement in replacements:
+        if replacement["name"] == replaced_by and \
+           replacement["version"] == replacement_version:
+            break
+    else:
+        raise Exception("Can not found expected replacement for the component"
+                        " {component} {version}".format(component=component,
+                                                        version=version))
+
+
+@then('I should find that the component {component} version {version} has only one replacement')
```

Comment:
IMHO we still can check both replacements, just the test would need to contain two checks:

* I should find that component A version 1.0.0 can be replaced by component Z version 10.0.0
* I should find that component B version 3.2.1 can be replaced by component Z version 10.0.0